src/libstd/io/stdio.rs

```diff
@@ -872,6 +872,92 @@ impl fmt::Debug for StderrLock<'_> {
     }
 }
 
+/// Locks the current handle and reads a line of input, returning a `String` containing
+/// the input.
+///
+/// If you need more explicit control over
+/// locking, see the [`Stdin::lock`] and [`Stdout::lock`] methods.
+///
+/// [`Stdin::lock`]: struct.Stdin.html#method.lock
+/// [`Stdout::lock`]: struct.Stdout.html#method.lock
+///
+/// ### Note: Safety and Usage
+/// This function is not the recommended solution to get user input, it should
+/// only be used for simple actions. If you require more explicit control over
+/// getting user input, see the [`Stdin::read_line`] method.
+///
+/// [`Stdin::read_line`]: struct.Stdin.html#method.read_line
+///
+/// # Examples
+///
+/// ```rust
+/// use std::io::input_prompt;
+///
+/// fn main() {
+///     let user_input = input_prompt("Please enter some text: ");
+///
+///     println!("You typed: {}", user_input);
+///
+/// }
+/// ```
+#[unstable(
+    feature = "input_prompt",
+    reason = "this function may be replaced with a more general mechanism",
+    issue = "none"
+)]
+pub fn input_prompt(prompt: &str) -> String {
+    stdout()
+        .write(prompt.as_bytes())
+        .unwrap();
+    return input();
+}
+
+/// Equivalent to the [`input_prompt`] method except that a prompt is not printed before
+/// capturing input.
+///
+/// [`input_prompt`]: #method.input_prompt
+///
+/// ### Note: Safety and Usage
+/// This function is not the recommended solution to get user input, it should
+/// only be used for simple actions. If you require more explicit control over
+/// getting user input, see the [`Stdin::read_line`] method.
+///
+/// [`Stdin::read_line`]: struct.Stdin.html#method.read_line
+///
+/// # Examples
+///
+/// ```rust
+/// use std::io::input;
+///
+/// fn main() {
+///     print!("Please enter some text: ");
+///
+///     let user_input = input();
+///
+///     println!("You typed: {}", user_input);
+///
+/// }
+/// ```
+#[unstable(
+    feature = "input",
+    reason = "this function may be replaced with a more general mechanism",
+    issue = "none"
+)]
+pub fn input() -> String {
+    let mut input = String::new();
+    let _ = stdout().flush();
+    stdin()
+        .read_line(&mut input)
+        .expect("Failed to read from stdin");
+    if input.chars().next_back().unwrap() == '\n' {
+        input.pop();
+    }
+    if input.chars().next_back().unwrap() == '\r' {
```

Comment:
Is it a good idea to do this on a non-windows machine? I wonder what is the behavior of `String::split_whitespace` on this.

Reply:
I changed the code and it should run "slightly" faster and "should" work on linux, unix and windows. I don't know about redox, since i don't know what escape sequence it uses to signify that the enter key was pressed .

Reply:
Slightly faster? Isn't not checking for `\r` at all faster in unix-based system (anything other than windows). Maybe use a platform guard?